main.py

```diff
@@ -123,65 +152,83 @@ def send_duml(s, source, target, cmd_type, cmd_set, cmd_id, payload = None):
 
     sequence_number += 1
 
-print('app version: 3.0.0\n')
+print('app version: 3.1.0\n')
 # Open serial.
 try:
-
-    result = []
-    ports = serial.tools.list_ports.comports(True)
-
-    for port in ports:
-        try:
-            print(port.description)
-            if port.description.find("For Protocol") != -1:
-                print("found DJI USB VCOM For Protocol")
-                s = serial.Serial(port=port.name, baudrate=115200)
-                print('Opened serial port:', s.name)
-            else:
-                print("skip")
-            result.append(port)
-        except (OSError, serial.SerialException):
-            pass
+    if args.port:
+        s = serial.Serial(port=args.port, baudrate=115200)
+        print('Opened serial port:', s.name)
+    else:
+        result = []
+        ports = serial.tools.list_ports.comports(True)
+
+        for port in ports:
+            try:
+                print(port.description)
+                if port.description.find("For Protocol") != -1:
+                    print("found DJI USB VCOM For Protocol")
+                    s = serial.Serial(port=port.name, baudrate=115200)
+                    print('Opened serial port:', s.name)
+                    break
+                else:
+                    print("skip")
+                result.append(port)
+            except (OSError, serial.SerialException):
+                pass
+        
+        try: s
+        except NameError: raise serial.SerialException("Unable to find the correct COM port.\nPerhaps try to supply the portname to the command by adding '-p <COM_PORT>'")
 
 except serial.SerialException as e:
     print('Could not open serial port:', e)
     exit(1)
 
 # Stylistic: Newline for spacing.
 print('\nDji RC231 emulation started...\n')
-print('\nClose terminal to stop\n')
+print('\nTo stop, press Ctrl-C or close the terminal\n')
 
 print('*******************************************************\n')
 print('* Telegram: https://t.me/DJI_RC_N1_SIMULATOR_FLY_DC   *\n')
 print('* Donate: https://www.buymeacoffee.com/ivanyakymenko  *\n')
 print('*******************************************************\n')
 
-# Process input (min 364, center 1024, max 1684) -> (min 0, center 16384, max 32768)
+# Process input (min 364, center 1024, max 1684)
+inr = [364, 1684]
+if args.driver == 'vgamepad': outr = [-32768, 32767] # min -32768, max 32767
+elif args.driver == 'vjoy': outr = [0, 32767] # min 0, max 32768
 def parseInput(input, name):
-    output = (int.from_bytes(input, byteorder='little') - 1024) * 2 * 4096 // 165
-    if output >= 32768:
-        output = 32767
+    # Source: https://stackoverflow.com/a/929107
+    output = (((int.from_bytes(input, byteorder='little') - inr[0]) * (outr[1] - outr[0])) / (inr[1] - inr[0])) + outr[0]
+
+    if name in invert:
+        if args.driver == 'vgamepad': output = -output
```

Comment:
```suggestion
        if args.driver == 'vgamepad': output
```

If adding "= -output" the controllers will get inverted creating a problem

Reply:
Have you read the lines ABOVE that line? 

` if name in invert:`

So basically it is INTENTED to be INVERTED!